packages/xdl/src/Android.ts

```diff
@@ -139,18 +139,32 @@ export async function installExpoAsync(url?: string) {
     total: 100,
     width: 40,
   });
-  const warningTimer = setTimeout(() => {
-    Logger.global.info(
-      'This download is taking longer than expected. You can also try downloading the clients from the website at https://expo.io/tools'
+
+  let warningTimer: NodeJS.Timeout;
+  const setWarningTimer = () => {
+    if (warningTimer) {
+      clearTimeout(warningTimer);
+    }
+    return setTimeout(
+      () =>
+        Logger.global.info(
+          'This download is taking longer than expected. You can also try downloading the clients from the website at https://expo.io/tools'
+        ),
+      INSTALL_WARNING_TIMEOUT
     );
-  }, INSTALL_WARNING_TIMEOUT);
+  };
+
   Logger.notifications.info({ code: NotificationCode.START_LOADING });
+  warningTimer = setWarningTimer();
```

Comment:
It's a bit weird that I have to set it here, couldn't do it in `setWarningTimer`. Mostly because TypeScript told me the value was uninitialised. 🙆‍♂